files/image_config/corefile_uploader/core_uploader.py

```diff
@@ -0,0 +1,294 @@
+
+    while len(e) != len(l):
+        e.append(l[len(e)])
+        os.mkdir("/".join(e))
+
+def make_new_dir(p):
+    if os.path.isdir(p):
+        rmtree(p)
+    elif os.path.exists(p):
+        os.path.unlink(p)
+    mkdir(p)
+
+def parse_a_json(data, prefix, val):
+    for i in data:
+        if type(data[i]) == dict:
+            parse_a_json(data[i], prefix + (i,), val)
+        else:
+            val[prefix + (i,)] = data[i]
+
+class config:
+    parsed_data = {}
+    cfg_data = {}
+
+    def __init__(self):
+        while not os.path.exists(RC_FILE):
+            # Wait here until service restart
+            log_err("Unable to retrieve Azure storage credentials")
+            time.sleep (HOURS_4)
+
+        with open(RC_FILE, 'r') as f:
+            self.parsed_data = json.load(f)
+            parse_a_json(self.parsed_data, (), self.cfg_data)
+
+    def get_data(self, k):
+        return self.cfg_data[k] if self.cfg_data.has_key(k) else ""
+
+    def get_dict(self):
+        return self.parsed_data
+
+    def get_core_info(self, corepath, devicename):
+        info = {}
+        info["corefname"] = os.path.basename(corepath)
+        info["tstamp"] = str(os.stat(corepath).st_ctime)
+        info["devicename"] = devicename
+
+        lpath = self.get_data(("metadata_files_in_archive", "core_info"))
+        f = open(lpath, "w+")
+        f.write(json.dumps(info))
+        f.close()
+
+        return lpath
+
+
+class Watcher:
+
+    def __init__(self):
+        self.observer = Observer()
+
+    def run(self):
+        event_handler = Handler()
+        self.observer.schedule(event_handler, CORE_FILE_PATH)
+        self.observer.start()
+        try:
+            while True:
+                time.sleep(5)
+        except:
+            self.observer.stop()
+            log_err("Error in watcher")
+
+        self.observer.join()
+
+def set_env(lst):
+    for k in lst:
+        if lst[k]:
+            os.environ[k] = lst[k]
+            log_debug("set env {} = {}".format(k, lst[k]))
+
+class Handler(FileSystemEventHandler):
+
+    @staticmethod
+    def init():
+        global hostname, sonicversion, asicname, acctname, acctkey, sharename
+        global cwd, cfg
+
+        cfg = config()
+
+        set_env(cfg.get_dict()["env"])
+
+        hostname = socket.gethostname()
+        if not hostname:
+            raise Exception("Failed to read hostname from /etc/sonic/sonic_version.yml")
+
+        acctname = cfg.get_data(("azure_sonic_core_storage", "account_name"))
+        acctkey = cfg.get_data(("azure_sonic_core_storage", "account_key"))
+        sharename = cfg.get_data(("azure_sonic_core_storage", "share_name"))
+
+        if not acctname or not acctkey or not sharename:
+            while True:
+                # Wait here until service restart
+                log_err("Unable to retrieve Azure storage credentials")
+                time.sleep (HOURS_4)
+
+        with open("/etc/sonic/sonic_version.yml", 'r') as stream:
```

Comment:
Why do you use absolute file path name when you defined it in metadata? Or why do you define 'version' in metadata file?

Reply:
sonic_version.yaml is in uploaded archive.
It is parsed as version & asictype are used in the remote dir path.